# Chocolate Ratings Analysis

## 1. Data Exploration

This notebook explores the raw chocolate ratings dataset before any data cleaning or transformation.

The goal of this stage is to understand:

- The size and structure of the dataset
- Available features
- Data types
- Missing values
- Duplicate records
- Basic numerical and categorical statistics

### Import Libraries

For the initial exploration, only Pandas is required.

In [11]:
import pandas as pd

### Load the Dataset

The original dataset is loaded directly from the `data/raw` directory.

The raw file will remain unchanged throughout the project.

In [12]:
DATA_PATH = "../data/raw/chocolate.csv"

df = pd.read_csv(DATA_PATH)

### Preview the Data

Let's inspect the first few rows to get an initial understanding of the dataset.

In [13]:
df.head()

,Company,Specific Bean\nOrigin,Review\nDate,Cocoa\nPercent,Company\nLocation,Rating,Bean\nType,Broad Bean\nOrigin
0,A. Morin,Agua Grande,2016,63%,France,3.75,Blend,Sao Tome
1,A. Morin,Kpime,2015,70%,France,2.75,Blend,Togo
2,A. Morin,Atsane,2015,70%,France,3.00,Blend,Togo
3,A. Morin,Akata,2015,70%,France,3.50,Blend,Togo
4,A. Morin,Quilla,2015,70%,France,3.50,Blend,Peru


In [14]:
df.shape

(1795, 8)

### Dataset Dimensions

The shape of the dataset shows the number of observations and features.

In [15]:
print(f"Rows: {df.shape[0]:,}")
print(f"Columns: {df.shape[1]}")

Rows: 1,795
Columns: 8


### Column Names

Before cleaning the dataset, let's inspect the original column names.

In [16]:
df.columns.tolist()

['Company',
 'Specific Bean\nOrigin',
 'Review\nDate',
 'Cocoa\nPercent',
 'Company\nLocation',
 'Rating',
 'Bean\nType',
 'Broad Bean\nOrigin']

### Dataset Information

This provides an overview of the column names, data types, and non-null values.

In [17]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1795 entries, 0 to 1794
Data columns (total 8 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Company               1795 non-null   str    
 1   Specific Bean
Origin  1795 non-null   str    
 2   Review
Date           1795 non-null   int64  
 3   Cocoa
Percent         1795 non-null   str    
 4   Company
Location      1795 non-null   str    
 5   Rating                1795 non-null   float64
 6   Bean
Type             1795 non-null   str    
 7   Broad Bean
Origin     1794 non-null   str    
dtypes: float64(1), int64(1), str(6)
memory usage: 112.3 KB


### Missing Values

Missing values can affect later analysis, so we check how many are present in each column.

In [18]:
missing_values = df.isna().sum()

missing_values

Company                  0
Specific Bean\nOrigin    0
Review\nDate             0
Cocoa\nPercent           0
Company\nLocation        0
Rating                   0
Bean\nType               0
Broad Bean\nOrigin       1
dtype: int64

In [19]:
missing_values[missing_values > 0]

Broad Bean\nOrigin    1
dtype: int64

### Duplicate Records

Duplicate rows can distort statistical analysis, so we check whether any exact duplicates exist.

In [20]:
duplicate_count = df.duplicated().sum()

print(f"Duplicate rows: {duplicate_count}")

Duplicate rows: 0


### Numerical Summary

Summary statistics help identify the scale, range, and distribution of numerical features.

In [21]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
Review\nDate,1795.0,2012.325348,2.927210,2006.0,2010.000,2013.00,2015.0,2017.0
Rating,1795.0,3.185933,0.478062,1.0,2.875,3.25,3.5,5.0


## Initial Observations

The raw dataset contains **1,795 rows and 8 columns**.

Initial inspection shows that:

- Several column names contain newline characters (`\n`) and should be standardized.
- Only one missing value is present, in the `Broad Bean Origin` column.
- No exact duplicate rows were found.
- Further inspection of column formats and data types will be performed during the cleaning stage.

Overall, the dataset appears relatively clean and suitable for further preparation and analysis.